vars/withOpenshiftServices.groovy

```diff
@@ -0,0 +1,124 @@
+def call(services, body) {
+    List<String> names = getNames(services)
+    try {
+        createOpenshiftResources(services, names)
+        withEnv(env(services, names)) {
+            body()
+        }
+    } finally {
+        cleanup(names)
+    }
+}
+
+def cleanup(names) {
+    for (int i = 0; i < names.size(); i++) {
+        String resourceName = names[0]
+        openshiftScale deploymentConfig: resourceName,  replicaCount: 0
+        openshift.withCluster() {
+            openshift.selector('svc', [name: resourceName]).delete()
+            openshift.selector('dc', [name: resourceName]).delete()
+        }
+    }
+}
+
+def createOpenshiftResources(services, names) {
+    Map jobs = [:]
+    for (int i = 0; i < services.size(); i++) {
+        String service = services[i]
+        String name = names[i]
+        jobs[name] = {
+            openshiftCreateResource(getDeploymentConfigYaml(service, name))
+            openshiftCreateResource(getServiceYaml(service, name))
+            openshiftScale deploymentConfig: name,  replicaCount: 1, verifyReplicaCount: 1
+        }
+    }
+    parallel jobs
+}
+
+String sanitizeObjectName(s) {
+    s.replace('_', '-').toLowerCase()
+}
+
+Map<String, String> getNames(services) {
+    List<String> names = []
+    for (int i = 0; i < services.size(); i++) {
+        names += sanitizeObjectName("${env.BUILD_TAG}-${services[i]}")
+    }
+    return names
+}
+
+List<String> env(services, names) {
+    List<String> out = []
+    if (services.contains('mongodb')) {
+        out.add("MONGODB_HOST=${names[services.indexOf('mongodb')]}")
+    }
+    return out
+}
```

Comment:
I'd suggest at some point we move these methods out of here and into utils since they might be useful in other pipeline functions at some point.